controllers/managedocs_controller.go

```diff
@@ -885,6 +909,45 @@ func (r *ManagedOCSReconciler) reconcileAlertRelabelConfigSecret() error {
 	return nil
 }
 
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller
+func (r *ManagedOCSReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling Prometheus Service")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.prometheusService, func() error {
+		if err := r.own(r.prometheusService); err != nil {
+			return err
+		}
+
+		desired := templates.PrometheusServiceTemplate.DeepCopy()
+		r.prometheusService.Spec.Ports = desired.Spec.Ports
+		r.prometheusService.Spec.Selector = map[string]string{
+			monLabelKey: monLabelValue,
+		}
```

Comment:
Is there a reason why we are not just forcing the entire template's spec and then mutating it?

```suggestion
		desired := templates.PrometheusServiceTemplate.DeepCopy()
		desired.Spec.Selector = map[string]string{
			monLabelKey: monLabelValue,
		}
		r.prometheusService.Spec = desired.Spec		
```

Reply:
Yes. As Service`s spec contains other items like `ClusterIP` etc, which we should not reset. So we are only reset the `Ports` part of the Spec.
Previously the code was as you suggested, which brought lot of test failures (like `spec.ClusterIP` is an immutable var and should not be set)

Reply:
So in that case there is no need for a template at all. Please remove hard code into the reconcile function

Reply:
Or alternatively, override the spec and than copy back the stuff that does not need to be overriden. 
We already used that pattern once in this controller

Reply:
No templates for Service then...